File: Utility/Utility.fs
module Utility

open System.Reflection
open System.IO

let public GetToken(key: string) = 
  let assembly = Assembly.GetExecutingAssembly()
  let stream = assembly.GetManifestResourceStream("Utility..env")
  let env = (new StreamReader(stream)).ReadToEnd()

  let index = env.IndexOf(key)
  let startIndex = env.IndexOf('\"', index) + 1;
  let endIndex = env.IndexOf('\"', startIndex);
  env[startIndex..endIndex - 1];

File: CrypticCode/Utility.cs
using System.Reflection;

namespace CrypticCode;

public static class Utility {
  public static string GetToken(string key) {
    Stream? stream = Assembly
      .GetExecutingAssembly()
      .GetManifestResourceStream("CrypticCode..env")
      ?? throw new DirectoryNotFoundException();
    string env = new StreamReader(stream)
      .ReadToEnd();

    int index = env.IndexOf(key);
    if(index < 0) return string.Empty;

    int startIndex = env.IndexOf('\"', index) + 1;
    int endIndex = env.IndexOf('\"', startIndex);
    return env[startIndex..endIndex];
  }
}
